plugins/BEdita/Core/src/Model/Table/UsersTable.php

```diff
@@ -247,6 +249,39 @@ protected function findMine(Query $query)
         });
     }
 
+    /**
+     * Common login conditions for direct and external auth login
+     *
+     * @return array Conditions to use in query
+     */
+    protected function loginConditions()
+    {
+        return [
+            $this->aliasField('deleted') => false,
+            $this->aliasField('blocked') => false,
+            $this->aliasField('status') . ' IN' => ['on', 'draft'],
+        ];
+    }
+
+    /**
+     * Finder for users login with `username`
+     * Valid attributes for `blocked`, `deleted` and `status` are checked
+     *
+     * @param \Cake\ORM\Query $query Query object instance.
+     * @return \Cake\ORM\Query
+     * @throws \Cake\Network\Exception\BadRequestException if `username` is missing
+     */
+    protected function findLogin(Query $query, array $options)
```

Comment:
I'd change this finder. It shouldn't be necessary to add the condition on username on our side: the auth component should already handle that, and in a more flexible way than this (the "username" field can be configured).

Therefore, I'd make this finder only filter _all_ users that are able to login:

```php
function findLogin(Query $query)
{
    return $query
        ->where(function (QueryExpression $exp) {
            return $exp
                ->eq($this->aliasField('deleted'), false)
                ->eq($this->aliasField('blocked'), false)
                ->in($this->aliasField('status'), ['on', 'draft']);
        });
}
```

Doing so, the `loginConditions()` method would become useless, and the `findExternalAuth()` finder could simply apply `$query->find('login')` to the query it manipulates.

Reply:
good point, I'm changing it

Reply:
Ok, changed but `username` still necessary.
Ended up with two finders.